packages/sdk/src/wasm/VerifiablePresentationV1/request.ts

```diff
@@ -0,0 +1,179 @@
+import { Buffer } from 'buffer/index.js';
+import _JB from 'json-bigint';
+
+import { sha256 } from '../../hash.js';
+import {
+    AccountAddress,
+    AccountSigner,
+    AccountTransaction,
+    AccountTransactionHeader,
+    AccountTransactionType,
+    ConcordiumGRPCClient,
+    NextAccountNonce,
+    RegisterDataPayload,
+    cborDecode,
+    cborEncode,
+    signTransaction,
+} from '../../index.js';
+import { ContractAddress, DataBlob, TransactionExpiry, TransactionHash } from '../../types/index.js';
+import { CredentialStatement, StatementProverQualifier } from '../../web3-id/types.js';
+import { GivenContextJSON, givenContextFromJSON, givenContextToJSON } from './internal.js';
+import { CredentialContextLabel, GivenContext } from './types.js';
+
+const JSONBig = _JB({ useNativeBigInt: true, alwaysParseAsBig: true });
+
+// NOTE: renamed from ContextInformation in ADR
+export type Context = {
+    type: 'ConcordiumContextInformationV1';
+    given: GivenContext[];
+    requested: CredentialContextLabel[];
+};
+
+export function createContext(context: Omit<Context, 'type'>): Context {
+    return { type: 'ConcordiumContextInformationV1', ...context };
+}
+
+export function createSimpleContext(nonce: Uint8Array, connectionId: string, contextString: string): Context {
+    return createContext({
+        given: [
+            { label: 'Nonce', context: nonce },
+            { label: 'ConnectionID', context: connectionId },
+            { label: 'ContextString', context: contextString },
+        ],
+        requested: ['BlockHash', 'ResourceID'],
+    });
+}
+
+export type AnchorData = {
+    type: 'CCDVRA';
+    version: number;
+    hash: Uint8Array;
+    public?: string;
+};
+
+export function createAnchor(
+    context: Context,
+    credentialStatements: CredentialStatement[],
+    publicInfo?: string
+): Uint8Array {
+    const hash = computeAnchorHash(context, credentialStatements);
+    const data: AnchorData = { type: 'CCDVRA', version: 1, hash, public: publicInfo };
+    return cborEncode(data);
+}
+
+export function computeAnchorHash(context: Context, credentialStatements: CredentialStatement[]): Uint8Array {
+    // TODO: this is a quick and dirty anchor implementation that needs to be replaced with
+    // the one from concordium-base when available.
```

Comment:
I don't think that base will calculate the anchor hash but I agree that we need to align in ADR (and different SDKs/wallets) how it is calculated.